src/res_core.ml

```diff
@@ -6452,3 +6442,13 @@ and parseExtension ?(moduleLanguage=false) p =
   let attrId = parseAttributeId ~startPos p in
   let payload = parsePayload p in
   (attrId, payload)
+
+(* module signature on the file level *)
+let parseSpecification p : Parsetree.signature =
+  parseRegion p ~grammar:Grammar.Specification ~f:parseSignatureItemRegion
+  [@@progress (Parser.next, Parser.expect, Parser.checkProgress)]
```

Comment:
These annotations go on recursive functions, or the check won't fire.
Moved to `parseSignatureItemRegion`  and `parseStructureItemRegion` in a PR.

Reply:
@cristianoc Thanks!

Reply:
Oops, sorry about merging so eagerly